weave/context_state.py

```diff
@@ -6,6 +6,46 @@
 from . import server_interface
 from . import uris
 
+# colab currently runs ipykernel < 6.0.  This resets context on every
+# execution, see: https://github.com/ipython/ipykernel/pull/632.  We
+# maintain a global context to work around this.
+patch_context = False
```

Comment:
consider adding a comment that this logic relies on the variable name being the name of the context var with a prefix of "_"